rust/operator-binary/src/spark_controller.rs

```diff
@@ -58,7 +59,8 @@ pub struct Ctx {
     pub product_config: ProductConfigManager,
 }
 
-#[derive(Snafu, Debug)]
+#[derive(Snafu, Debug, EnumDiscriminants)]
+#[strum_discriminants(derive(IntoStaticStr))]
 #[allow(clippy::enum_variant_names)]
 pub enum Error {
     #[snafu(display("object {obj_ref} is missing metadata to build owner reference"))]
```

Comment:
Nit: These `ObjectRef<SparkCluster>`s are not really needed anymore, since `Controller` now provides that context itself.

Reply:
Ah that is awesome thanks. Will adapt.